core/pubnub_crypto.c

```diff
@@ -115,7 +115,7 @@ int pubnub_encrypt(char const* cipher_key, pubnub_bymebl_t msg, char* base64_str
     #endif
 
     int max_size = base64_max_size(encrypted.size);
-    if (*n + 1 < max_size) {
+    if (*n + 1 < (size_t)max_size) {
```

Comment:
This is ok, but maybe the `base64_max_size` function definition should change to return `size_t` to avoid these casts. I feel like `size_t` is a more appropriate type for allocation sizes than `int`.

The only reason I wrote it to return `int` is because it accepts an `int` input, matching the `base64encode` definition, whose two size inputs could also be changed to `size_t`. The `int` return value on `base64encode` is for error handling.

https://github.com/pubnub/c-core/blob/4a5896634b91ef7ca4ec8676d2d090d664173556/core/pubnub_crypto.h#L145-L153